tutorials/fundamentals/add-to-array.md

```diff
@@ -0,0 +1,93 @@
+JavaScript arrays have 3 methods for adding an element to an array:
+
+- [`push()`](https://www.w3schools.com/jsref/jsref_push.asp) adds to the end of the array
+- [`unshift()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift) adds to the beginning of the array
+- [`splice()`](/tutorials/fundamentals/array-splice) adds to the middle of the array
+
+Below are examples of using `push()`, `unshift()`, and `splice()`.
+
+```javascript
+[require:Fundamentals array append push$]
+```
+
+```javascript
+[require:Fundamentals array append unshift$]
+```
+
+```javascript
+[require: Array#splice.*add]
+```
+
+These methods modify the array _in place_, which means they modify `arr` rather than creating a [copy of `arr`](/tutorials/fundamentals/copy-array).
+You can also use the [spread operator](https://thecodebarbarian.com/object-assign-vs-object-spread.html) and other [immutable methods](/tutorials/fundamentals/array-append#immutable-methods) that create a new array and leave `arr` unmodified.
+
+```javascript
+[require:Fundamentals array append concat$]
+```
+
+Setting an Index Directly
+-------------------------
+
+If you're adding an element to the end of the array, you don't necessarily have to use `push()`.
+You can just set the array index, and JavaScript will update the array's length for you.
+
+```javascript
+[require:Fundamentals array append direct set]
+```
+
+JavaScript does **not** throw an error if you set an out of bounds array index.
+For example, if your array has length 3 and you set index `4`, JavaScript will just grow your array by adding a [hole in the array](https://2ality.com/2015/09/holes-arrays-es6.html).
+
+```javascript
+const arr = ['a', 'b', 'c'];
+
+arr[4] = 'e';
+
+arr.length; // 5
+arr; // [ 'a', 'b', 'c', <1 empty item>, 'e' ]
```

Comment:
Should you also mention that if they iterate through the array they'll have an entry return undefined which might cause problems?